src/lib/util/multibyte.h

```diff
@@ -0,0 +1,163 @@
+// license:BSD-3-Clause
+// copyright-holders:Mietek Bak
+
+#ifndef MAME_LIB_UTIL_MULTIBYTE_H
+#define MAME_LIB_UTIL_MULTIBYTE_H
+
+#pragma once
+
+#include <stdint.h>
+
+//////////////////////////////////////////////////////////////////////////////
+
+// unsigned big-endian
+
+inline const uint16_t get_uint16be(const uint8_t *const buf)
```

Comment:
I think the principle of defining these helpers for endian-specific numeric I/O is good, the specific implementation and approach needs to be reviewed and sanctioned by @cuavas. Once that's done, probably the other nscsi code (and probably elsewhere in MAME such as chd) should be reviewed and updated to use them as well.

Reply:
Meanwhile, I’ll note there are many places in MAME that could use multibyte accessors, and some that already do.  For example:
https://github.com/mamedev/mame/blob/a4e202ae77cc605ad1a8bc006d0b9d80e8952ccf/src/lib/util/aviio.cpp#L577-L715
https://github.com/mamedev/mame/blob/a4e202ae77cc605ad1a8bc006d0b9d80e8952ccf/src/tools/imgtool/modules/ti990hd.cpp#L53-L75

Reply:
This sort of thing shows up in the filesystem code as well. I did several experiments with utility functions and classes to help deal with this sort of thing (the `sext` function I added to `src/lib/util/coretmpl.h` was a byproduct of one of those), but was not quite convinced that both the API or the implementation details were as correct as they needed to be for the whole MAME project.

Reply:
I’ll add 64-bit accessors to make `"multibyte.h"` easier to reuse in the future.